app/assets/javascripts/discourse/app/components/sidebar/user/custom-sections.js

```diff
@@ -2,30 +2,53 @@ import Component from "@glimmer/component";
 import { inject as service } from "@ember/service";
 import { ajax } from "discourse/lib/ajax";
 import { bind } from "discourse-common/utils/decorators";
-import Section from "discourse/components/sidebar/user/section";
+import Section from "discourse/lib/sidebar/section";
+import CommunitySection from "discourse/lib/sidebar/community-section";
 
 export default class SidebarUserCustomSections extends Component {
   @service currentUser;
   @service router;
   @service messageBus;
+  @service appEvents;
+  @service topicTrackingState;
+  @service siteSettings;
 
   constructor() {
     super(...arguments);
     this.messageBus.subscribe("/refresh-sidebar-sections", this._refresh);
+    this.cacheSections;
   }
 
   willDestroy() {
     this.messageBus.unsubscribe("/refresh-sidebar-sections");
+    return this.cacheSections.forEach((section) => {
```

Comment:
I had to introduce cacheSection because willDestroy was calling sections constructor and was creating new topic tracking state callbacks

Reply:
I think this means `sections()` should not be a getter and can be a normal function that is called once and assigned to a field in the `constructor`.

Reply:
I tried it as a first approach. We want to have sections tracked, that when you update section, state is updated. But we don't want to create new instances on willDestroy

Reply:
Ic. In this case, I believe we can cache the getter?

```
import { cached } from "@glimmer/tracking";

@cached
get sections() {
```